modules/internal/String.chpl

```diff
@@ -427,6 +427,75 @@ module String {
       return ret;
     }
 
+    /* This function takes an array of stings as input.
+
+    returns:  The longest common prefix path of all the input paths
+    */
+    proc commonPrefix(paths: []): string {
+
+    var prefix: string = "";    // result string
+    var inputLength = paths.size;   // size of input array
+    var firstPath = paths[1];
+    var delimiter: string;
+    var cnt: int = 1;
+    var pos: int ;    // rightmost index of common prefix
+    var minPathLength: int ;
+
+    // if input is empty, return empty string.
+    // if input is just one string, return that string as longest common prefix path.
+
+    if inputLength == 0 then {
+      return prefix;
+    } else if inputLength == 1 then{
+      return firstPath;
+    }
+
+    // finding delimiter to split the paths.
+
+    if firstPath.find("\\", 1..firstPath.length) == 0 then {
+      delimiter = "/";
+    } else {
+      delimiter = "\\";
+    }
+    cnt = firstPath.count(delimiter);
+
+    var prefixArray:[1..cnt+1] string;    // array of resultant prefix string
+    prefixArray = firstPath.split(delimiter, -1, false);
```

Comment:
You could just declare this all in one go (and wouldn't need to specify the exact array being created), but since the line would get a bit long it's up to you.
```chapel
var prefixArray = firstPath.split(delimiter, -1, false);
```

Reply:
This would remove the need for calling `count` above, as then you could just get `cnt` from the size of the prefix array created

Reply:
(Or even make it a local variable to the `for` loop below)

Reply:
Thanks for mentioning this feature.